src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+package mate.academy.dao.impl;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String COLUMN_ID = "id";
+    private static final String COLUMN_TITLE = "title";
+    private static final String COLUMN_PRICE = "price";
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books(title, price) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setInt(2, book.getPrice().intValue());
+            int executedRows = preparedStatement.executeUpdate();
+            if (executedRows < 1) {
+                throw new DataProcessingException("Expected to insert at least 1 row "
+                        + "but inserted 0 rows");
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();

```

Comment:
ResultSet it's also implements AutoCloseable, so we need also use in this place try with resources